app/pages/ReporterTypePage.scala

```diff
@@ -84,18 +84,43 @@ case object ReporterTypePage extends QuestionPage[ReporterType] {
     SecondContactEmailPage,
     SecondContactHavePhonePage,
     SecondContactPhonePage,
-    RegistrationInfoPage
+    WhatIsYourNamePage,
+    IndDoYouHaveNINumberPage
   )
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "reporterType"
 
-  override def cleanup(value: Option[ReporterType], userAnswers: UserAnswers): Try[UserAnswers] = value match {
-    case Some(Individual)                                              => individualCleanup.foldLeft(Try(userAnswers))(PageLists.removePage)
-    case Some(Sole)                                                    => soleTraderTypeCleanup.foldLeft(Try(userAnswers))(PageLists.removePage)
-    case Some(reporterType) if orgReporterTypes.contains(reporterType) => otherBusinessTypeCleanup.foldLeft(Try(userAnswers))(PageLists.removePage)
-    case _                                                             => super.cleanup(value, userAnswers)
-  }
+  override def cleanup(value: Option[ReporterType], userAnswers: UserAnswers): Try[UserAnswers] =
+    value match {
+      case Some(Individual) =>
+        cleanupPages(userAnswers, individualCleanup)
+          .flatMap(registrationInfoCleanup(_, Individual))
+
+      case Some(Sole) =>
+        cleanupPages(userAnswers, soleTraderTypeCleanup)
+          .flatMap(registrationInfoCleanup(_, Sole))
+
+      case Some(reporterType) if orgReporterTypes.contains(reporterType) =>
```

Comment:
Hello @C-Thomp  Quick Question -> is this if condition orgReporterTypes.contains(reporterType) required?

Because Sole & Individual are already covered in above case so remaining ones are orgReporterTypes.

Reply:
Hi @jayandran-Sampath, I think you are correct however i think it's worth being for each case for readability of the code

Reply:
cool.. just thought of avoiding unnecessary nesting.. :)